source/h2_stream.c

```diff
@@ -815,7 +812,13 @@ int aws_h2_stream_encode_data_frame(
             &connection->thread_data.window_size_peer,
             output,
             &input_stream_complete,
-            &input_stream_stalled)) {
+            &input_stream_stalled,
+            &input_stream_error)) {
+
+        if (input_stream_error) {
+            /* If the error comes from the input stream, propagate it to the current_write */
+            current_write->error_code = aws_last_error();
```

Comment:
(continuing previous comment)
so like here, we could pop the current_write out of its list and destroy it